src/service_router.rs

```diff
@@ -104,71 +104,77 @@ pub fn create<T: Controller>(controller: T) -> Chain {
-        assert_eq!(result, "request processed");
-    }
+            let result = response::extract_body_to_string(response);
+            assert_eq!(result, "request processed");
+        }
 
-    it "should return an error if no service was found" {
-        let response = request::get("http://localhost:3000/unknown-id/a-command",
-                        Headers::new(),
-                        &service_router).unwrap();
+        it "should return an error if no service was found" {
+            let response = request::get("http://localhost:3000/unknown-id/a-command",
+                            Headers::new(),
+                            &service_router).unwrap();
 
-        let result = response::extract_body_to_string(response);
-        assert_eq!(result, "No Such Service: unknown-id");
+            let result = response::extract_body_to_string(response);
+            assert_eq!(result, "No Such Service: unknown-id");
+        }
     }
 
-    it "should get the appropriate CORS headers" {
-        use iron::headers;
-        use super::CORS;
+    describe! cors {
+        before_each {
+            use iron::Headers;
+            use controller::FoxBox;
+            use iron_test::request;
 
-        for endpoint in CORS::ENDPOINTS {
-            let (_, path) = *endpoint;
-            let path = "http://localhost:3000/".to_owned() +
-                       &(path.join("/").replace("*", "foo"));
-            match request::options(&path, Headers::new(), &service_router) {
-                Ok(res) => {
-                    let headers = &res.headers;
-                    assert!(headers.has::<headers::AccessControlAllowOrigin>());
-                    assert!(headers.has::<headers::AccessControlAllowHeaders>());
-                    assert!(headers.has::<headers::AccessControlAllowMethods>());
-                },
-                _ => {
-                    assert!(false)
+            let controller = FoxBox::new(false, Some("localhost".to_owned()), None, None);
+            let service_router = create(controller.clone());
+        }
+
+        it "should get the appropriate CORS headers" {
+            use iron::headers;
+            use super::super::CORS;
+
+            for endpoint in CORS::ENDPOINTS {
+                let (_, path) = *endpoint;
+                let path = "http://localhost:3000/".to_owned() +
+                           &(path.join("/").replace("*", "foo"));
+                match request::options(&path, Headers::new(), &service_router) {
+                    Ok(res) => {
+                        let headers = &res.headers;
+                        assert!(headers.has::<headers::AccessControlAllowOrigin>());
+                        assert!(headers.has::<headers::AccessControlAllowHeaders>());
+                        assert!(headers.has::<headers::AccessControlAllowMethods>());
+                    },
+                    _ => {
+                        assert!(false)
```

Comment:
I think we can make this part smaller:
```rust
                let response = request::options(&path, Headers::new(), &service_router).unwrap();
                let headers = &response.headers;
                assert!(headers.has::<headers::AccessControlAllowOrigin>());
                assert!(headers.has::<headers::AccessControlAllowHeaders>());
                assert!(headers.has::<headers::AccessControlAllowMethods>());
```

If the unwrap() fails, the thread will panic and the test will fail. More over, the error message will be more detailed than `panicked at 'assertion failed: false'`